core/lib/src/fairing/ad_hoc.rs

```diff
@@ -61,22 +61,34 @@ pub struct AdHoc {
 //     );
 // }
 
+/// The attach fairing's function type
```

Comment:
I don't think this is worth the noise. If we ever get trait aliases. then we can do this in a much nicer way that gets more reuseability.

Reply:
I've added this allow :
`#[allow(clippy::type_complexity)] // Private enum, the types will be simplified if needed.`